tests/test_rpms.py

```diff
@@ -576,16 +579,61 @@ def test_log_cmd_with_retries(capsys):
     logger.addHandler(handler)
 
     with patch('time.sleep') as mocked_sleep:
-        with patch('subprocess.Popen.wait', side_effect=[1,1,1,0]) as mocked_wait:
+        with patch('subprocess.Popen.wait', side_effect=[3,2,1,0]) as mocked_wait:
             assert_that(calling(processor.log_cmd).with_args(['echo', 'hello'], tries=4), exits(0))
             assert len(mocked_wait.mock_calls) == 4
             assert mocked_sleep.call_args_list == [call(30), call(60), call(90)]
 
     out, err = capsys.readouterr()
-
     # should fail three times and succeed in the forth time
-    expected = '[WARNING]  Command echo hello failed, will retry in 90s [tried: 3/4]'
-    assert expected in err
+    expected = \
+        ['[WARNING] Command echo hello failed, exit code: 3. Will retry in 30s [tried: 1/4]',
+         '[WARNING] Command echo hello failed, exit code: 2. Will retry in 60s [tried: 2/4]',
+         '[WARNING] Command echo hello failed, exit code: 1. Will retry in 90s [tried: 3/4]']
+
+    for expected_item in expected:
+        assert expected_item in err
+
+
+@pytest.mark.parametrize('fatal', [True, False])
+def test_log_cmd_with_retries_failure(capsys, fatal):
+    remove_handlers()
+    capsys.readouterr()
+
+    mock_options = MagicMock(koji=False)
+    with patch('os.path.isfile', return_value=True):
+        processor = BaseProcessor(mock_options)
+
+    logger = logging.getLogger('altsrc')
+    handler = logging.StreamHandler()
+    handler.setLevel('DEBUG')
+    handler.setFormatter(logging.Formatter('%(asctime)s [%(levelname)s] %(message)s'))
+    logger.addHandler(handler)
+
+    with patch('time.sleep') as mocked_sleep:
+        with patch('subprocess.Popen.wait', side_effect=lambda: 1) as mocked_wait:
+            if fatal:
+                assert_that(calling(processor.log_cmd).with_args(['echo', 'hello'], fatal=fatal,
```

Comment:
Just want to note that you don't have to duplicate all that calling(...) like this.  calling() returns some object, like anything else, so it could be written like:

```
calling_log = calling(...).with_args(...)
if fatal:
  assert_that(calling_log, raises(CommandError))
else:
  assert_that(calling_log, exits(0))
```